### DISCUSSION TITLE
Postscript failure

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Ed/Community#299] [Installation] [answered]
Hi All,

CSP: GCP
Sunbird DevOps: `release-5.1.0`

As we running postscript jenkins jobs getting the following error

`fatal: [localhost]: FAILED! => {
    "access_control_allow_origin": "*",
    "changed": false,
    "connection": "close",
    "content": "{\"id\":\"api.org.create\",\"ver\":\"v1\",\"ts\":\"2022-12-09 06:44:37:988+0000\",\"params\":{\"resmsgid\":\"ea43930ea2e237240d14252e90b6d747\",\"msgid\":\"ea43930ea2e237240d14252e90b6d747\",\"err\":\"UOS_ORGCRT0018\",\"status\":\"FAILED\",\"errmsg\":\"Invalid organisationType: board. Valid values are: [].\"},\"responseCode\":\"CLIENT_ERROR\",\"result\":{}}",
    "content_length": "319",
    "content_type": "application/json",
    "date": "Fri, 09 Dec 2022 06:44:37 GMT",
    "elapsed": 0,
    "json": {
        "id": "api.org.create",
        "params": {
            "err": "UOS_ORGCRT0018",
            "errmsg": "Invalid organisationType: board. Valid values are: [].",
            "msgid": "ea43930ea2e237240d14252e90b6d747",
            "resmsgid": "ea43930ea2e237240d14252e90b6d747",
            "status": "FAILED"
        },
        "responseCode": "CLIENT_ERROR",
        "result": {},
        "ts": "2022-12-09 06:44:37:988+0000",
        "ver": "v1"
    },
    "redirected": false,
    "referrer_policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
    "status": 400,
    "url": "https://uphrh.in/api/org/v1/create",
    "via": "kong/0.14.1",
    "x_content_type_options": "nosniff",
    "x_frame_options": "DENY",
    "x_kong_proxy_latency": "20",
    "x_kong_upstream_latency": "118",
    "x_permitted_cross_domain_policies": "master-only",
    "x_ratelimit_limit_hour": "5000",
    "x_ratelimit_remaining_hour": "4999",
    "x_xss_protection": "1; mode=block"
}

MSG:

Status code was 400 and not [200]: HTTP Error 400: Bad Request`

**Comment by beepdot** [comment, marked as answer]
@reshmi-nair Can you please provide your inputs here?

**Reply by tsprasath** [reply]
@keshavprasadms @reshmi-nair Executed via postman following error faced and learner service logs attached.

`curl --location --request POST 'https://uphrh.in/api/org/v1/create' \
--header 'x-authenticated-user-token: xxxxxxxxxxxx' \
--header 'Authorization: Bearer xxxxxxxxxxxxxxxxxxxx' \
--header 'Content-Type: application/json' \
--data-raw '{
          "request": {
            "orgName": "Sunbird",
            "channel": "sunbird",
            "description": "Default Channel",
            "isTenant": true,
            "organisationType":"board"
          }
        }'`
Postman ERROR
`{
    "id": "api.org.create",
    "ver": "v1",
    "ts": "2022-12-09 10:43:33:835+0000",
    "params": {
        "resmsgid": "0ba547440de3dc17df645f12dd78aa44",
        "msgid": "0ba547440de3dc17df645f12dd78aa44",
        "err": "UOS_ORGCRT0018",
        "status": "FAILED",
        "errmsg": "Invalid organisationType: board. Valid values are: []."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}`

**Learner error logs**
`stamp":"2022-12-09T10:23:55.226Z","msg":"{\"eid\":\"ERROR\",\"ets\":1670581435226,\"ver\":\"3.0\",\"mid\":\"233b0a858a0f534d8ca03a1ae64a3a95\",\"actor\":{\"id\":\"a58c6e88-f5a6-4098-ae14-41a486aa5e61\",\"type\":\"User\"},\"context\":{\"channel\":\"ORG_001\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.0.0\"},\"env\":\"Organisation\",\"cdata\":[{\"id\":\"233b0a858a0f534d8ca03a1ae64a3a95\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"err\":\"UOS_ORGCRT0018\",\"stacktrace\":\"Invalid organisationType: board. Valid values are: []. org.sunbird.actor.organisation.validator.OrganisationRequestValidator.validateOrgType(OrganisationRequestValidator.java:124)org.sunbird.actor.organisation.OrganisationManagementActor.createOrg(OrganisationManagementActor.java:80)org.sunbird.actor.organisation.OrganisationManagementActor.onReceive(OrganisationManagementActor.java:46)org.sunbird.actor.core.BaseActor.onReceive(BaseActor.java:21)akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:291)akka.actor.Actor$class.aroundReceive(Actor.scala:517)akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:180)akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)akka.actor.ActorCell.invoke(ActorCell.scala:557)akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)akka.dispatch.Mailbox.run(Mailbox.scala:225)akka.dispatch.Mailbox.exec(Mailbox.scala:235)akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\",\"errtype\":\"api_access\",\"requestid\":\"233b0a858a0f534d8ca03a1ae64a3a95\"}}","lname":"TelemetryEventLogger","tname":"Thread-166","level":"INFO","HOSTNAME":"learner-849d9c8986-jqtfx","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2022-12-09T10:23:55.227Z","msg":"{\"eid\":\"LOG\",\"ets\":1670581435227,\"ver\":\"3.0\",\"mid\":\"233b0a858a0f534d8ca03a1ae64a3a95\",\"actor\":{\"id\":\"a58c6e88-f5a6-4098-ae14-41a486aa5e61\",\"type\":\"User\"},\"context\":{\"channel\":\"ORG_001\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.0.0\"},\"env\":\"Organisation\",\"cdata\":[{\"id\":\"233b0a858a0f534d8ca03a1ae64a3a95\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"level\":\"info\",\"type\":\"Api_access\",\"message\":\"{eid='LOG', edata={level=trace, requestid=233b0a858a0f534d8ca03a1ae64a3a95, type=system, message=EXIT LOG: method : POST, url: /v1/org/create , For Operation : createOrg, params=[{msgid=233b0a858a0f534d8ca03a1ae64a3a95, errmsg=Invalid organisationType: board. Valid values are: []., resmsgid=233b0a858a0f534d8ca03a1ae64a3a95, err=UOS_ORGCRT0018, status=FAILED, responseCode=400}]}}\",\"params\":[]}}","lname":"TelemetryEventLogger","tname":"Thread-166","level":"INFO","HOSTNAME":"learner-849d9c8986-jqtfx","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}`

**Reply by tsprasath** [reply]
@keshavprasadms @reshmi-nair Issued related to Cassandra migration has not been properly executed. Manually executed all the tables are updated.
Reference related files from

`https://github.com/Sunbird-Lern/sunbird-utils/tree/release-4.9.0_RC1/sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra`

The below command will help you to execute a file in Cassandra.

`cqlsh < V1.100_cassandra.cql'`

And re-ran the Postscript.


Troubleshooting steps:

Login to cassandra
`select * from sunbird.organisation;`

![Screenshot from 2022-12-09 20-08-06](https://user-images.githubusercontent.com/20708426/206726201-8aab7af5-0596-4292-a802-caac96f22c74.png)

Above are the expected.

Note: Restart the learner service in Kubernetes after updating the tables. Else it will throw old data from the table.

**Reply by beepdot** [reply]
Do you need any more information on this from us or is the issue resolved?

**Reply by tsprasath** [reply]
@keshavprasadms  still facing issues with content and taxonomy after deploying the service and need to re-deploy the postscript and let you know.

**Reply by sukhpreetssekhon** [reply]
@tsprasath - Please share the exact issue here so that @keshavprasadms can debug and support you.

**Reply by tsprasath** [reply]
@keshavprasadms @sukhpreetsamagra I am facing an issue with ` Configuration error: Configuration error[/home/sunbird/taxonomy-service-1.0-SNAPSHOT/config/application.conf: 371: Expecting a value but got wrong token: 'newline' (JSON does not allow unescaped newline in quoted strings, use a backslash escape) (if you intended 'newline' (JSON does not allow unescaped newline in quoted strings, use a backslash escape) to be part of a key or string value, try enclosing the key or value in double quotes)]`

**Reply by beepdot** [reply]
Lets not mix the issues in multiple threads and duplicate the responses. The issue you mentioned is tracked here - https://github.com/orgs/Sunbird-Ed/discussions/298

**Reply by tsprasath** [reply]
@keshavprasadms  Fixed content and taxonomy service Post script started running
